micrometer-core/src/main/java/io/micrometer/core/instrument/push/PushMeterRegistry.java

```diff
@@ -50,14 +53,30 @@ protected PushMeterRegistry(PushRegistryConfig config, Clock clock) {
      * Catch uncaught exceptions thrown from {@link #publish()}.
      */
     private void publishSafely() {
-        try {
-            publish();
+        if (this.publishing.compareAndSet(false, true)) {
+            try {
+                publish();
+            }
+            catch (Throwable e) {
+                logger.warn("Unexpected exception thrown while publishing metrics for " + getClass().getSimpleName(),
+                        e);
+            }
+            finally {
+                this.publishing.set(false);
+            }
         }
-        catch (Throwable e) {
-            logger.warn("Unexpected exception thrown while publishing metrics for " + getClass().getSimpleName(), e);
+        else {
+            logger.warn("Publishing is already in progress.");
```

Comment:
It would be good to also explicitly state that the publish will consequently be skipped.